waku/v2/node/waku_node.nim

```diff
@@ -491,43 +491,36 @@ proc unsubscribe*(node: WakuNode, pubsubTopic: PubsubTopic, contentTopics: Conte
 ## Waku archive
 
 proc mountArchive*(node: WakuNode,
-                   driver: Option[ArchiveDriver],
-                   messageValidator: Option[MessageValidator],
-                   retentionPolicy: Option[RetentionPolicy]) =
+                   driver: ArchiveDriver,
+                   retentionPolicy = none(RetentionPolicy)):
+                   Result[void, string] =
 
-  if driver.isNone():
-    error "failed to mount waku archive protocol", error="archive driver not set"
-    return
-
-  node.wakuArchive = WakuArchive.new(driver.get(), messageValidator, retentionPolicy)
-
-# TODO: Review this periodic task. Maybe, move it to the appplication code
-const WakuArchiveDefaultRetentionPolicyInterval* = 30.minutes
-
-proc executeMessageRetentionPolicy*(node: WakuNode) =
-  if node.wakuArchive.isNil():
-    return
+  let wakuArchiveRes = WakuArchive.new(driver,
+                                       retentionPolicy)
+  if wakuArchiveRes.isErr():
+    return err("error in mountArchive: " & wakuArchiveRes.error)
 
-  debug "executing message retention policy"
+  node.wakuArchive = wakuArchiveRes.get()
 
   try:
-    waitFor node.wakuArchive.executeMessageRetentionPolicy()
-    waitFor node.wakuArchive.reportStoredMessagesMetric()
+    let reportMetricRes = waitFor node.wakuArchive.reportStoredMessagesMetric()
+    if reportMetricRes.isErr():
+      return err("error in mountArchive: " & reportMetricRes.error)
   except CatchableError:
-    debug "Error executing retention policy " & getCurrentExceptionMsg()
+    return err("exception in mountArchive: " & getCurrentExceptionMsg())
 
-proc startMessageRetentionPolicyPeriodicTask*(node: WakuNode, interval: Duration) =
-  if node.wakuArchive.isNil():
-    return
-
-  # https://github.com/nim-lang/Nim/issues/17369
-  var executeRetentionPolicy: proc(udata: pointer) {.gcsafe, raises: [Defect].}
-  executeRetentionPolicy = proc(udata: pointer) {.gcsafe.} =
-    executeMessageRetentionPolicy(node)
-    discard setTimer(Moment.fromNow(interval), executeRetentionPolicy)
+  if retentionPolicy.isSome():
+    try:
+      debug "executing message retention policy"
+      let retPolRes = waitFor node.wakuArchive.executeMessageRetentionPolicy()
+      if retPolRes.isErr():
+        return err("error in mountArchive: " & retPolRes.error)
+    except CatchableError:
+      return err("exception in mountArch-ret-pol: " & getCurrentExceptionMsg())
 
-  discard setTimer(Moment.fromNow(interval), executeRetentionPolicy)
+    node.wakuArchive.startMessageRetentionPolicyPeriodicTask(30.minutes)
```

Comment:
Nit: Better to extract this to a `const` just before the function than to make this a "magic value" inline with the code: easier to debug; clear where to add config if needed.